### PR TITLE
feat: Add findLastIndex for Array and TypedArray (Stage 3)

### PR DESCRIPTION
**Pull request by MaxGraey**
Also some refactoring in Array.ts and TypedArray.ts for consistency

- [x] I've read the contributing guidelines

### CODE REVIEW

File: std/portable/index.d.ts [hunk 1]
```diff
@@ -312,57 +312,132 @@ declare interface StringConstructor {
   fromCodePoints(arr: i32[]): string;
 }
 
+declare interface String {
+  /** Returns value using relative indexing. Index may be negative */
+  at(index: i32): string;
+}
+
```
Comment (dcodeIO) [inline std/portable/index.d.ts:319]: In `portable` we are reusing the TS-provided definition of `String`. Does is "merge" with its definition and add to it, or what is the idea here?

Comment (MaxGraey) [reply std/portable/index.d.ts:319]: Yes, I think that with this method in the original interface they should merge without any conflicts

Comment (MaxGraey) [reply std/portable/index.d.ts:319]: https://www.typescriptlang.org/play?ts=4.4.0-beta#code/MYGwhgzhAEDKAuAnaBTAHvFA7AJjBiAllgObQDeAUNDdGPABTE7oBc0WArgLYBGKiAJTsISYmSq0p0RCnidEWaPAAWhCAG1m6ALoBuarQC+lE5Ragws6MUyIAZmGAo4SCoZr0muNhx78hETFSAyMgA

**Comment by MaxGraey** [comment]
btw we nave problem with unavailable nightly build again

**Comment by dcodeIO** [comment]
It sometimes becomes available a bit later, may be that it just hasn't finished the build yet. Let's see :)

**Comment by MaxGraey** [comment]
yeah

**Pull request merged**